geode-core/src/test/java/org/apache/geode/internal/cache/versions/RegionVersionVectorTest.java

```diff
@@ -567,7 +572,56 @@ public void testRecordVersionAfterRegionInitThrowsException() {
     rvv.recordVersion(ownerId, tag);
   }
 
-  public RegionVersionVector createRegionVersionVector(InternalDistributedMember ownerId,
+  @Test
+  public void usesNewVersionIfGreaterThanOldVersion() throws Exception {
+    VersionSource<InternalDistributedMember> ownerId = mock(VersionSource.class);
+    long oldVersion = 1;
+    long newVersion = 2;
+
+    RegionVersionVector rvv = new TestableRegionVersionVector(ownerId, oldVersion);
+    rvv.updateLocalVersion(newVersion);
+    assertThat(rvv.getVersionForMember(ownerId)).isEqualTo(newVersion);
+  }
+
+  @Test
+  public void usesOldVersionIfGreaterThanNewVersion() throws Exception {
+    VersionSource<InternalDistributedMember> ownerId = mock(VersionSource.class);
+    long oldVersion = 2;
+    long newVersion = 1;
+
+    RegionVersionVector rvv = new TestableRegionVersionVector(ownerId, oldVersion);
+    rvv.updateLocalVersion(newVersion);
+    assertThat(rvv.getVersionForMember(ownerId)).isEqualTo(oldVersion);
+  }
+
+  @Test
+  public void doesNothingIfVersionsAreSame() throws Exception {
+    VersionSource<InternalDistributedMember> ownerId = mock(VersionSource.class);
+    long oldVersion = 2;
+    long sameVersion = 2;
+
+    RegionVersionVector rvv = new TestableRegionVersionVector(ownerId, oldVersion);
+    rvv.updateLocalVersion(sameVersion);
+    assertThat(rvv.getVersionForMember(ownerId)).isEqualTo(oldVersion);
+  }
+
+  @Test
+  public void doesNotHangIfOtherThreadChangedVersion() throws Exception {
+    VersionSource<InternalDistributedMember> ownerId = mock(VersionSource.class);
+    long oldVersion = 1;
+    long newVersion = 2;
+
+    RegionVersionVector rvv = new VersionRaceConditionRegionVersionVector(ownerId, oldVersion);
+    Future<Boolean> future = executor.submit(() -> {
```

Comment:
I can't understand why this test needs the executor. It does not seem that another thread does anything.
Wouldn't this test do the same validation if you just had the test method directly call rvv.updateLocalVersion?

Reply:
For tests, you can avoid the need for creating and managing your own executor by using CompletableFuture (which takes either an executor or uses the default ForkJoinPool, which is perfectly good for most tests).So for example, `CompletableFuture.runAsync(() -> myMethod())`

Reply:
I understand now that you did not want the test to hang forever if it does hang.
I think the test method name "doesNotHangIfOtherThreadChangedVersion" mislead me.
It made me think the test used another thread to change the version. But you actually doing it inline in a single thread. So I suggest you rename this test to not mention threads. Something like "doesNotHangIfVersionChangesDuringUpdateLocalVersion".